cli/bench_command.go

```diff
@@ -66,305 +62,540 @@ type benchCmd struct {
 	deDuplicationWindow  time.Duration
 	retries              int
 	retriesUsed          bool
+	ack                  bool
 }
 
 const (
-	DefaultDurableConsumerName string = "natscli-bench"
+	DefaultDurableConsumerName string = "nats-bench"
 	DefaultStreamName          string = "benchstream"
 	DefaultBucketName          string = "benchbucket"
+	DefaultServiceName         string = "nats-bench-service"
+	DefaultServiceVersion      string = "1.0.0"
+	BenchTypeCorePub           string = "pub"
+	BenchTypeCoreSub           string = "sub"
+	BenchTypeServiceRequest    string = "request"
+	BenchTypeServiceServe      string = "reply"
+	BenchTypeJSPub             string = "jspub"
+	BenchTypeJSOrdered         string = "jsordered"
+	BenchTypeJSConsume         string = "jsconsume"
+	BenchTypeJSFetch           string = "jsfetch"
+	BenchTypeOldJSOrdered      string = "oldjsordered"
+	BenchTypeOldJSPush         string = "oldjspush"
+	BenchTypeOldJSPull         string = "oldjspull"
+	BenchTypeKVPut             string = "kvput"
+	BenchTypeKVGet             string = "kvget"
+	AckModeNone                string = "none"
+	AckModeAll                 string = "all"
+	AckModeExplicit            string = "explicit"
 )
 
 func configureBenchCommand(app commandHost) {
 	c := &benchCmd{}
 
 	benchHelp := `
-Core NATS publish and subscribe:
+Core NATS publish:
 
-  nats bench benchsubject --pub 1 --sub 10
+  nats bench pub benchsubject
 
-Request reply with queue group:
+Core NATS subscribe:
```

Comment:
nice its split into commands but I think we should focus on the help on each sub command keeping with the overall style of the CLI being more list with short description above.  If the individual commands are well documented and have good descriptions it largely remove the need for this

```
Subcommands:
  stream add       Create a new Stream
  stream ls        List all known Streams
  stream report    Reports on Stream statistics
  stream find      Finds streams matching certain criteria
  stream info      Stream information
```

Then expand each commands help individually with more details and perhaps even add some examples into `nats cheat`

Reply:
Reworked the help and updated the cheat sheet file as well. Indeed it is now much better with the sub commands now there's no need to provide examples in the in-line help (but they are there in the cheats sheet file).